### PR TITLE
Add supplementary source example

### LINKED ISSUE #153: Example of using SupplementaryView
Can you provide example how to use SupplementaryView? Will be great if you could apply SupplementaryView to first item in Mixed data example. cc @rawlinxx, @Sherlouk

### PR DESCRIPTION
**Pull request by rnystrom**
## Changes in this pull request

Got around to adding a supplementary view source example.

Fixes #153 #163

img_0197

## Pull request checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: Example/IGListKitExamples/Models/FeedItem.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import IGListKit
+
+class FeedItem: IGListDiffable {
```
Comment (jessesquires) [inline Example/IGListKitExamples/Models/FeedItem.swift]: final

File: Example/IGListKitExamples/SectionControllers/FeedItemSectionController.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import IGListKit
+
+class FeedItemSectionController: IGListSectionController, IGListSectionType, IGListSupplementaryViewSource {
```
Comment (jessesquires) [inline Example/IGListKitExamples/SectionControllers/FeedItemSectionController.swift]: final

File: Example/IGListKitExamples/ViewControllers/SupplementaryViewController.swift [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import UIKit
+import IGListKit
+
+class SupplementaryViewController: UIViewController, IGListAdapterDataSource {
```
Comment (jessesquires) [inline Example/IGListKitExamples/ViewControllers/SupplementaryViewController.swift]: final

File: Example/IGListKitExamples/Views/UserHeaderView.swift [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import UIKit
+
+class UserHeaderView: UICollectionViewCell {
```
Comment (jessesquires) [inline Example/IGListKitExamples/Views/UserHeaderView.swift]: final

**Comment by rnystrom** [comment]
@HumidBrains if you're using `UICollectionViewFlowLayout` then all you have to do is set [sectionHeadersPinToVisibleBounds](https://developer.apple.com/reference/uikit/uicollectionviewflowlayout/1617699-sectionheaderspintovisiblebounds) to `YES` and you're good!

Within Instagram we hyper optimize and have written our own `UICollectionViewLayout` with sticky-header logic. The gist of it is to return `NO` when the `bounds.origin` changes in `shouldInvalidateLayoutForBoundsChange:` then in your `layoutAttributesForSupplementaryViewOfKind:atIndexPath:` do some math adjusting the frame origin of the attributes so they "stick".

I've seen some open source examples of this, but if you can, I'd just start off w/ `UICollectionViewFlowLayout`.

**Comment by HumidBrains** [comment]
Thanks! And what layout framework are you recommending to use in conjunction with IGListKit and Swift? I saw someone asking about AsyncDisplayKit, but to me it looks like the benefits from using ASDisplayNode goes lost?

**Pull request closed without merging**